### PR TITLE
CI: apps

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

test App as standlaone

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda @tchaton @rohitgr7

### CODE REVIEW

**Comment by Borda** [comment]
@tchaton @manskx any idea why this tests is failing?
```
________________ test_prepare_lightning_wheels_and_requirement _________________

tmpdir = local('/tmp/pytest-of-runner/pytest-0/test_prepare_lightning_wheels_0')

    def test_prepare_lightning_wheels_and_requirement(tmpdir):
        """This test ensures the lightning source gets packaged inside the lightning repo."""
    
        cleanup_handle = _prepare_lightning_wheels_and_requirements(tmpdir)
        from lightning_app.__version__ import version
    
        tar_name = f"lightning-app-{version}.tar.gz"
        loaded_files = sorted(os.listdir(tmpdir))
>       assert loaded_files[0] == tar_name
E       IndexError: list index out of range
```

File: tests/tests_app/utilities/packaging/test_lightning_utils.py [hunk 1]
```diff
@@ -11,16 +11,17 @@
...
-    """This test ensures the lightning source gets packaged inside the lightning repo."""
-
-    cleanup_handle = _prepare_lightning_wheels_and_requirements(tmpdir)
-    from lightning.__version__ import version
-
-    tar_name = f"lightning-{version}.tar.gz"
-    assert sorted(os.listdir(tmpdir))[0] == tar_name
-    cleanup_handle()
-    assert os.listdir(tmpdir) == []
+# TODO: this is very sensitive test, need to be done
```
Comment (Borda) [inline tests/tests_app/utilities/packaging/test_lightning_utils.py:14]: Discussed with @ethanfurman that we are fine to drop it
cc: @tchaton @manskx

**Review by ethanwharris — APPROVED** [review]
As mentioned offline, the failing test can be fixed by extending the packaging to support `lightning_app` package in addition to just the `lightning` package. But this can be done in a follow-up :smiley:

File: tests/tests_app/utilities/packaging/test_lightning_utils.py [hunk 2]
```diff
@@ -10,17 +9,17 @@
     _verify_lightning_version,
 )
 
-
-def test_prepare_lightning_wheels_and_requirement(tmpdir):
```
Comment (awaelchli) [inline tests/tests_app/utilities/packaging/test_lightning_utils.py:14]: I think we should keep the test

File: src/lightning_app/utilities/packaging/lightning_utils.py [hunk 3]
```diff
@@ -83,6 +83,8 @@ def get_dist_path_if_editable_install(project_name) -> str:
     """Is distribution an editable install - modified version from pip that
     fetches egg-info instead of egg-link"""
```
Comment (awaelchli) [inline src/lightning_app/utilities/packaging/lightning_utils.py:84]: could we ask @tchaton for review on this one?

**Review by awaelchli — APPROVED** [review]

File: src/lightning_app/utilities/packaging/lightning_utils.py (hunk 3 above)
Comment (Borda) [reply src/lightning_app/utilities/packaging/lightning_utils.py:84]: I did, so pending now :)

File: tests/tests_app/utilities/packaging/test_lightning_utils.py (hunk 2 above)
Comment (Borda) [reply tests/tests_app/utilities/packaging/test_lightning_utils.py:14]: Any idea how to fix it? We can open a PR with uncommenting this test right after this is merged...

**Review by tchaton — APPROVED** [review]
LGTM! But cloud dev might be broken in the meanwhile. Let see.

**Review by akihironitta — APPROVED** [review]

File: .azure/app-cloud-e2e.yml [hunk 4]
```diff
@@ -101,17 +101,14 @@ jobs:
...
+    - bash: git restore . && python -m pip install -e .[ui] --find-links https://download.pytorch.org/whl/cpu/torch_stable.html
       env:
         PACKAGE_NAME: app
+      displayName: 'Install lightning app'
 
-    - bash: git restore . && python -m pip install -e . --find-links https://download.pytorch.org/whl/cpu/torch_stable.html
-      displayName: 'Install pytorch lightning'
+    - bash: python -m pip install -r requirements/app/test.txt
       env:
         PACKAGE_NAME: pytorch
```
[the remark this answers is not in the document]
Comment (Borda) [reply .azure/app-cloud-e2e.yml:111]: it does as the PL is here installed from source so this env. var specify the scope
if as todo stated we will have as test dependency PL from PyPI we can drop it

File: tests/tests_app/utilities/packaging/test_lightning_utils.py (hunk 2 above)
Comment (awaelchli) [reply tests/tests_app/utilities/packaging/test_lightning_utils.py:14]: Why did we leave the test commented out though? If we need to skip a test, we would add the skip mark instead. This is would be visible in the test output summary.

**Pull request merged**